yarn-project/aztec-nr/aztec/src/oracle/get_block_header.nr

```diff
@@ -13,34 +13,44 @@ use crate::{
     },
 };
 
+// TODO(#3564) - Nuke this oracle and Inject the number directly to context
+#[oracle(getNullifierRootBlockNumber)]
+fn get_nullifier_root_block_number_oracle(_nullifier_tree_root: Field) -> Field {}
+
+unconstrained pub fn get_nullifier_root_block_number(nullifier_tree_root: Field) -> u32 {
+    get_nullifier_root_block_number_oracle(nullifier_tree_root) as u32
+}
+
 #[oracle(getBlockHeader)]
-fn get_block_header_oracle(_block_number: Field) -> [Field; BLOCK_HEADER_LENGTH] {}
+fn get_block_header_oracle(_block_number: u32) -> [Field; BLOCK_HEADER_LENGTH] {}
 
-unconstrained pub fn get_block_header_internal(block_number: Field) -> BlockHeader {
+unconstrained pub fn get_block_header_internal(block_number: u32) -> BlockHeader {
     let block_header = get_block_header_oracle(block_number);
     BlockHeader::deserialize(block_header)
 }
 
-pub fn get_block_header(block_number: Field, context: PrivateContext) -> BlockHeader {
-    // 1) Get block header of a given block from oracle
+pub fn get_block_header(block_number: u32, context: PrivateContext) -> BlockHeader {
```

Comment:
this function name reads weirdly given the comment in the next line.
Looking at the fn name, I assume we want to get a block header from the block number. but the next comment does the opposite

Reply:
Not sure how we should do the wording much different. Its ok to me, but I might be stupid and have too much context.